misc/webrtc-utils/src/stream.rs

```diff
@@ -231,19 +246,33 @@ where
                     })?;
                     self.state.close_write_message_sent();
 
+                    // Set deadline when sending FIN
+                    self.fin_ack_deadline = Some(Instant::now() + FIN_ACK_TIMEOUT);
+
                     continue;
                 }
                 Some(Closing::MessageSent) => {
                     ready!(self.io.poll_flush_unpin(cx))?;
 
-                    self.state.write_closed();
-                    let _ = self
-                        .drop_notifier
-                        .take()
-                        .expect("to not close twice")
-                        .send(GracefullyClosed {});
+                    // Check if we've received FIN_ACK or deadline has passed
+                    if self.state.fin_ack_received()
```

Comment:
Not sure about this, but don't we need to somehow continue reading here to be able to receive the `FIN_ACK`? 

With `AsyncWriteExt` for instance a future will be created that only returns after `poll_close` succeeded i.e. no concurrent call to `poll_read` happens. With the current impl, I don't think a `FIN_ACK` will ever be read in that case?

On the other hand, we'd then have to discard the read messages or buffer them. Not sure how this is handled e.g. in QUIC, or in the go implementation.

Reply:
So I looked into the QUIC and TCP transports to check how they handle this in their `poll_close` function.
However, as far as I can tell neither of them wait in `poll_close` for the sent `FIN` to be acknowledged in the first palce..

Unfortunately I couldn't find any prior discussion about this.